pkg/k8shandler/fluentd.go

```diff
@@ -514,6 +499,61 @@ func (clusterRequest *ClusterLoggingRequest) updateFluentdDaemonsetIfRequired(de
 	return nil
 }
 
+func (clusterRequest *ClusterLoggingRequest) getFluentdAnnotations(daemonset *apps.DaemonSet) (map[string]string, error) {
+
+	if daemonset.Spec.Template.ObjectMeta.Annotations == nil {
+		daemonset.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
+	}
+
+	annotations := daemonset.Spec.Template.ObjectMeta.Annotations
+
+	fluentdTrustBundle := &v1.ConfigMap{}
+	fluentdTrustBundleName := types.NamespacedName{Name: constants.FluentdTrustedCAName, Namespace: constants.OpenshiftNS}
+	if err := clusterRequest.client.Get(context.TODO(), fluentdTrustBundleName, fluentdTrustBundle); err != nil {
+		if !errors.IsNotFound(err) {
+			return annotations, err
+		}
+	}
+
+	if _, ok := fluentdTrustBundle.Data[constants.TrustedCABundleKey]; !ok {
+		return annotations, fmt.Errorf("%v does not yet contain expected key %v", fluentdTrustBundle.Name, constants.TrustedCABundleKey)
+	}
+
+	trustedCAHashValue, err := calcTrustedCAHashValue(fluentdTrustBundle)
+	if err != nil {
+		return annotations, fmt.Errorf("unable to calculate trusted CA value. E: %s", err.Error())
+	}
+
+	if trustedCAHashValue == "" {
+		return annotations, fmt.Errorf("Did not receive hashvalue for trusted CA value")
```

Comment:
Does this need to return an error?  Isn't an empty value valid?

Reply:
it isn't, we want require that the hash is populated with something so that we don't unnecessarily update and cause a redeploy.